ballerina/channel_csv_io.bal

```diff
@@ -40,30 +40,56 @@ Error {
     return (check getReadableCSVChannel(readableChannel, 0)).csvStream();
 }
 
-isolated function channelWriteCsv(WritableChannel writableChannel, string[][]|map<anydata>[] content) returns Error? {
-    WritableCSVChannel csvChannel = check getWritableCSVChannel(writableChannel);
-    if content is string[][] {
-        foreach string[] r in content {
+isolated function channelWriteCsv(string path, FileWriteOption inputOption, string[][]|map<anydata>[] contentToWrite) returns Error? {
+    WritableCSVChannel csvChannel;
+    FileWriteOption option = OVERWRITE;
+    if contentToWrite is string[][] {
+        csvChannel = check getWritableCSVChannel(check openWritableCsvFile(path, option = inputOption));
+        foreach string[] r in contentToWrite {
             Error? writeResult = csvChannel.write(r);
             if writeResult is Error {
                 check csvChannel.close();
                 return writeResult;
             }
         }
-    } else if content is map<anydata>[] {
-        foreach map<anydata> row in content {
+        check csvChannel.close();
+    } else if contentToWrite is map<anydata>[] {
+        string[] headers = [];
+        if contentToWrite.length() == 0 {
+            return;
+        }
+        headers = contentToWrite[0].keys();
+        if inputOption == APPEND {
+            string[] headersFromCSV = check readHeadersFromCsvFile(path);
+            if headersFromCSV.length() > 0 {
+                headers = check validateCsvContent(headersFromCSV, headers);
+                headers = headersFromCSV;
+                option = APPEND;
```

Comment:
Should we override the fileOption manually? Yes the behaviour is the same in some cases of append but I still think file write option should not be changed.

Reply:
IMO better code would be to have separate functions,
1. Write to file when file is not found/empty
2. Append to file with records

This would be much clearer to read and concise. This is what you are trying to achieve by using an option variable arent you?

Reply:
Yes, to keep the writing part out of the if statements, we have to use a local variable named `option`. Otherwise we have to transfer the part back in to each if else statement which will reduce the readability

Reply:
My point is dont use if clause,
1. Read files and get headers
2. IF empty,
          writeContentWithHeaders()
     else {
           if option overwite,
                writeContentWithHeaders()
           else 
                // Validate headers
               // Append content in order w/o headers
      }

Reply:
Much better flow would be,
```
If option == OVERWRITE
   writeContentWithHeaders()
else
   readCSVFileAndGetHeaders()
   if headers not found (file not found/empty records)
         writeContentWithHeaders()
   else 
         // Validate headers
         // Append content in order w/o headers
}
```

Reply:
You have first done the headers validation then you go to write content. This should go hand in hand.

Reply:
Yes thats possible. WDYT @daneshk we will have to call writing functions back inside with each validation.

Reply:
yes, let's follow this, move them to separate functions

Reply:
This was updated in [161c3c5](https://github.com/ballerina-platform/module-ballerina-io/pull/827/commits/161c3c5941fe22f2de83b374115f83b32a63b6fb) and [300d1f2](https://github.com/ballerina-platform/module-ballerina-io/pull/827/commits/300d1f2c50d47725a3047f89e4671117f17fb9f3)